extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -95,104 +97,104 @@ public OpenCensusProtobufInputRowParser withParseSpec(ParseSpec parseSpec)
       Set<String> recordDimensions = metric.getMetricDescriptor().getLabelKeysList().stream()
           .map(s -> s.getKey())
           .collect(Collectors.toSet());
+
+      // Add resource map key set to record dimensions. Do not check for null behavior of
+      // getResource() and getLabelsMap() as default instance or empty map will be returned
+      // respectively. Hence it's safe to call methods directly without null check.
+      recordDimensions.addAll(metric.getResource().getLabelsMap().keySet());
+
+      // NAME, VALUE dimensions will not be present in labelKeysList or Metric.Resource map as they
+      // are derived dimensions, which get populated while parsing data for timeSeries hence add
+      // them to recordDimensions.
       recordDimensions.add(NAME);
       recordDimensions.add(VALUE);
 
-
       dimensions = Lists.newArrayList(
           Sets.difference(recordDimensions, parseSpec.getDimensionsSpec().getDimensionExclusions())
       );
     }
 
+    // Create immutable resourceLabels which are common to all points in timeSeries. It's safe to
+    // call getResource() and getLabelsMap() methods without null check.
+    Map<String, String> resourceLabels = ImmutableMap.copyOf(metric.getResource().getLabelsMap());
```

Comment:
Why is this copy of the map needed?  Can't you just do this down below on line 126?

```java
Map<String, Object> labels = new HashMap<>();
labels.addAll(metric.getResource().getLabelsMap());
```

Reply:
Agreed, the labels map itself is unmodifiable hence no need to create a copy. Done.